src/BIAnalyticsService/CreateParticipantScreeningEpisode/CreateParticipantScreeningEpisode.cs

```diff
@@ -0,0 +1,108 @@
+using System.Net;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Extensions.Logging;
+using System.Text.Json;
+using NHS.ServiceInsights.Common;
+using NHS.ServiceInsights.Model;
+
+namespace NHS.ServiceInsights.BIAnalyticsService;
+
+public class CreateParticipantScreeningEpisode
+{
+    private readonly ILogger<CreateParticipantScreeningEpisode> _logger;
+    private readonly IHttpRequestService _httpRequestService;
+    public CreateParticipantScreeningEpisode(ILogger<CreateParticipantScreeningEpisode> logger, IHttpRequestService httpRequestService)
+    {
+        _logger = logger;
+        _httpRequestService = httpRequestService;
+    }
+
+    [Function("CreateParticipantScreeningEpisode")]
+    public async Task<HttpResponseData> Run([HttpTrigger(AuthorizationLevel.Anonymous, "get")] HttpRequestData req)
+    {
+        _logger.LogInformation("Create Participant Screening Episode function start");
+
+        string episodeId = req.Query["EpisodeId"];
+
+        if (string.IsNullOrEmpty(episodeId))
+        {
+            _logger.LogError("episodeId is null or empty.");
+            return req.CreateResponse(HttpStatusCode.BadRequest);
+        }
+
+        var baseUrl = Environment.GetEnvironmentVariable("GetEpisodeUrl");
+        var getEpisodeUrl = $"{baseUrl}?EpisodeId={episodeId}";
+        _logger.LogInformation("Requesting episode URL: {Url}", getEpisodeUrl);
+
+        Episode episode;
+
+        try
+        {
+            var response = await _httpRequestService.SendGet(getEpisodeUrl);
+
+            if (!response.IsSuccessStatusCode)
+            {
+                _logger.LogError($"Failed to retrieve episode with Episode ID {episodeId}. Status Code: {response.StatusCode}");
+                return req.CreateResponse(response.StatusCode);
+            }
+
+            string episodeJson;
+            episodeJson = await response.Content.ReadAsStringAsync();
+            _logger.LogInformation("Episode data retrieved");
+            episode = JsonSerializer.Deserialize<Episode>(episodeJson);
+        }
+
+        catch (Exception ex)
+        {
+            _logger.LogError("Issue when getting episode from {getEpisodeUrl}. \nException: {ex}", getEpisodeUrl, ex);
```

Comment:
Update the comment to make clear that either retrieval or deserialisation failed